tests/test_iet.py

```diff
@@ -296,6 +297,52 @@ def _C_free(self):
 }"""
 
 
+def test_make_cuda_tensor_map():
```

Comment:
Note: this test is exciting many of the changes  in this PR. More tests in the accompanying PRO PR